cmake/GoogleCloudCppLibrary.cmake

```diff
@@ -22,6 +22,11 @@
 # * GOOGLE_CLOUD_CPP_SERVICE_DIRS: a list of service directories within the
 #   library.
 #
+# The following conditional arguments can be supplied to handle edge cases:
```

Comment:
Can this be an argument via `cmake_parse_arguments()`?

Reply:
Yes. Done.

This was a bit tricky to document. Hopefully it is clear.

Note that I am still setting `DOXYGEN_EXTRA_INCLUDES` in the parent scope. I do not intend to add that as an argument or document it in the helper function.